Isolate the rendering of the schema to just where it's required

ajmaddaford opened the pull request:
### What is the context of this PR?
Simplifying rendering before starting the 'Populate household composition blocks with full names' story. Moved the rendering of the schema so that it's only done when we're rendering the summary screen (where it's required so that piping works on the summary screen). 

### How to review 
Run tests

benjefferies commented:
@ajmaddaford can you rebase your branch with master or cherry pick your commit on top of master

iwootten committed:
Isolate the rendering of the schema to just where it's required

Moved the rendering of the schema so that it's only done when we're rendering the summary screen. This needs to be done so that piping works on the summary screen.
